isthmus/src/test/java/io/substrait/isthmus/NestedStructQueryTest.java

```diff
@@ -0,0 +1,427 @@
+                  child {
+                    struct_field {
+                      field: 1 # a
+                    }
+                  }
+                }
+              }
+              root_reference: {}
+            }
+        """;
+
+    test(table, query, expectedExpressionText);
+  }
+
+  @Test
+  public void testList2() {
+    final Table table =
+        new AbstractTable() {
+          @Override
+          public RelDataType getRowType(RelDataTypeFactory factory) {
+            var helper = new TypeHelper(factory);
+
+            return helper.struct2(
+                "x",
+                helper.i32(),
+                "a",
+                helper.list(helper.list(helper.list(helper.list(helper.i32())))));
+          }
+        };
+
+    String query =
+        """
+           SELECT
+             "nested"."my_table"."a"[1][2][3]
+           FROM
+             "nested"."my_table";
+           """;
+
+    String expectedExpressionText =
+        """
+        selection {
+          direct_reference {
+            list_element {
+              offset: 3
+              child {
+                list_element {
+                  offset: 2
+                  child {
+                    list_element {
+                      offset: 1
+                      child: {
+                        struct_field {
+                          field: 1 # a
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+          root_reference: {}
+        }
+        """;
+
+    test(table, query, expectedExpressionText);
+  }
+
+  @Test
+  public void testMap() throws SqlParseException {
+    final Table table =
+        new AbstractTable() {
+          @Override
+          public RelDataType getRowType(RelDataTypeFactory factory) {
+            var helper = new TypeHelper(factory);
+            return helper.struct(
+                "a", helper.map(helper.string(), helper.struct("c", helper.i32())));
+          }
+        };
+
+    String query =
+        """
+           SELECT
+             "nested"."my_table"."a"['foo']."c"
+           FROM
+             "nested"."my_table";
+           """;
+
+    String expectedExpressionText =
+        """
+            selection {
+              direct_reference {
+                struct_field {
+                  field: 0 # a
+                  child: {
+                    map_key: {
+                      map_key: {
+                        string: 'foo'
+                      }
+                      child: {
+                        struct_field: {
+                          field: 0 # c
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+              root_reference: {}
+            }
+        """;
+
+    test(table, query, expectedExpressionText);
+  }
+
+  @Test
+  public void testProtobufDoc() throws SqlParseException {
```

Comment:
What does this test verify that the other tests don't already?

Reply:
It's more an example from the doc: https://github.com/substrait-io/substrait/blob/main/proto/substrait/algebra.proto#L960-L966.

I could remove the previous step and keep this one.